exporter/azuredataexplorerexporter/adx_exporter.go

```diff
@@ -205,8 +206,19 @@ func getMappingRef(config *Config, telemetryDataType int) ingest.FileOption {
 }
 
 func buildAdxClient(config *Config, version string) (*kusto.Client, error) {
-	kcsb := kusto.NewConnectionStringBuilder(config.ClusterURI).WithAadAppKey(config.ApplicationID, string(config.ApplicationKey), config.TenantID)
-	kcsb.SetConnectorDetails("OpenTelemetry", version, "", "", false, "", kusto.StringPair{})
+	var kcsb *kusto.ConnectionStringBuilder
+	isManagedIdentity := len(strings.TrimSpace(config.ManagedIdentityID)) > 0
+	isSystemManagedIdentity := strings.EqualFold(strings.TrimSpace(config.ManagedIdentityID), "SYSTEM")
+	// If the user has managed identity done, use it. For System managed identity use the MI as system
+	switch {
+	case !isManagedIdentity:
+		kcsb = kusto.NewConnectionStringBuilder(config.ClusterURI).WithAadAppKey(config.ApplicationID, string(config.ApplicationKey), config.TenantID)
+	case isManagedIdentity && isSystemManagedIdentity:
+		kcsb = kusto.NewConnectionStringBuilder(config.ClusterURI).WithSystemManagedIdentity()
+	case isManagedIdentity && !isSystemManagedIdentity:
+		kcsb = kusto.NewConnectionStringBuilder(config.ClusterURI).WithUserManagedIdentity(config.ManagedIdentityID)
+	}
+	kcsb.SetConnectorDetails("OpenTelemetry", version, "", "", false, "", kusto.StringPair{Key: "isManagedIdentity", Value: strconv.FormatBool(isManagedIdentity)})
```

Comment:
This switch statement is weird to me, can you add tests to cover it?

Reply:
Added some tests for this